packages/toolboxes/evm/src/toolbox/BaseEVMToolbox.ts

```diff
@@ -461,6 +461,51 @@ const sendTransaction = async (
   }
 };
 
+const isAssetValueApproved = (
+  provider: Provider,
+  assetValue: AssetValue,
+  from: string,
+  spenderAddress: string,
+) => {
+  const { address } = assetValue;
+
+  if (!(address && from && typeof spenderAddress === "string")) {
+    throw new Error("core_approve_asset_address_or_from_not_found");
+  }
+  return isApproved(provider, {
+    amount: assetValue.getBaseValue("bigint"),
+    assetAddress: address,
+    from,
+    spenderAddress,
+  });
+};
+
+const approveAssetValue = (
+  provider: Provider,
+  assetValue: AssetValue,
+  signer?: Signer,
+  isEIP1559Compatible = true,
+  from?: string,
+  spenderAddress?: string,
+) => {
+  const { address } = assetValue;
+
+  if (!(address && from && typeof spenderAddress === "string")) {
+    throw new Error("core_approve_asset_address_or_from_not_found");
+  }
+  return approve(
```

Comment:
Xdefi Wallet overwrites the approve method. 

Could remove that method from the xdefi wallet integration, and test if that works?

Reply:
Same as with isAssetValueApproved - try to merge it with approve